code/modules/buildmode/submodes/variable_edit.dm

```diff
@@ -0,0 +1,57 @@
+/datum/buildmode_mode/varedit
+	key = "edit"
+	// Varedit mode
+	var/varholder = null
+	var/valueholder = null
+
+/datum/buildmode_mode/varedit/show_help(mob/user)
+	to_chat(user, "<span class='notice'>***********************************************************</span>")
+	to_chat(user, "<span class='notice'>Right Mouse Button on buildmode button = Select var(type) & value</span>")
+	to_chat(user, "<span class='notice'>Left Mouse Button on turf/obj/mob      = Set var(type) & value</span>")
+	to_chat(user, "<span class='notice'>Right Mouse Button on turf/obj/mob     = Reset var's value</span>")
+	to_chat(user, "<span class='notice'>***********************************************************</span>")
+
+/datum/buildmode_mode/vareedit/Reset()
+	. = ..()
+	varholder = null
+	valueholder = null
+
+/datum/buildmode_mode/varedit/change_settings(mob/user)
+	varholder = input(user,"Enter variable name:" ,"Name", "name")
+	
+	if(!vv_varname_lockcheck(varholder))
+		return
+
+	var/temp_value = user.client.vv_get_value()
```

Comment:
Is this a suitable interfacing with the existing VV system? Without an existing datum to interface with, this is the closest I could get. I made sure to split out the varname locking logic into its own proc, so I could use it here without accidentally allowing unauthorized access if things got changed later.

Also, it looks like the edit var COMSIG doesn't get fired in `object.vv_edit_var`. Should I fire that signal manually in here? Doing so feels like a bad idea.

Reply:
That signal is meant to trigger once per admin var edit. We only use it for a jokey plush atm which complains when var edits happen.

In other words no don't do anything with it.

Reply:
This is fine, important part is validation.